src/materials/ConcreteExpansionMicrocrackingDamage.C

```diff
@@ -0,0 +1,170 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteExpansionMicrocrackingDamage.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("BlackBearApp", ConcreteExpansionMicrocrackingDamage);
+
+template <>
+InputParameters
+validParams<ConcreteExpansionMicrocrackingDamage>()
+{
+  InputParameters params = validParams<ScalarDamageBase>();
+  params.addClassDescription("Scalar damage model based on extent of internal expansion");
+
+  params.addRequiredParam<MaterialPropertyName>(
+      "microcracking_eigenstrain_name",
+      "Name of the eigenstrain driving the microcracking damage process");
+
+  params.addParam<bool>("assume_isotropic_expansion",
+                        true,
+                        "Indicates whether the model assumes an isotropic expansion (true) or "
+                        "computes the linear expansion based on the first principal eigenstrain "
+                        "(false)");
+
+  params.addParam<bool>("include_confinement_effects",
+                        true,
+                        "Indicates whether the damage is affected by the current stress state");
+
+  params.addParam<Real>("eigenstrain_factor",
+                        1.0,
+                        "Correction factor by which the eigenstrain is multiplied before "
+                        "evaluating the damage");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "microcracking_initiation_strain",
+      "microcracking_initiation_strain > 0",
+      "Linear strain at which the microcracking initiates (in [m/m])");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "microcracking_strain_branch",
+      "microcracking_strain_branch > 0",
+      "Parameter controlling the rate at which the microcracking increases (in [m/m])");
+
+  params.addRangeCheckedParam<Real>(
+      "expansion_stress_limit",
+      "expansion_stress_limit > 0",
+      "Upper bound compressive stress beyond which damage is controlled by the external stress");
```

Comment:
The second argument to addRangeCheckedParam() is a default value, but you're missing that here. Since you want this model to work with any unit system, I don't think you want to have a default value. Since you can't have it default to 0 (because it would break the range check), I'd probably switch to addParam(), and then if 'include_confinement_effects' is true, require that this be specified, and in the constructor code enforce that it is greater than zero.